sling.go

```diff
@@ -517,17 +514,50 @@ func (s *Sling) doDecode(req *http.Request, successV, failureV interface{}) (*ht
 // decoding is skipped.
 // Caller is responsible for closing the resp.Body.
 func decodeResponse(resp *http.Response, decoder ResponseDecoder, successV, failureV interface{}) error {
-	if code := resp.StatusCode; 200 <= code && code <= 299 {
+	if code := resp.StatusCode; isSuccessful(code) {
 		if successV != nil {
 			return decoder.Decode(resp, successV)
 		}
 	} else {
 		if failureV != nil {
 			return decoder.Decode(resp, failureV)
 		}
-		var buf bytes.Buffer
-		buf.ReadFrom(resp.Body)
-		return fmt.Errorf(buf.String())
+
+		body, err := readWithCap(resp.Body, 100)
+		if err != nil {
+			return fmt.Errorf("status code %d was not successful and could not get body: %w", code, err)
+		}
```

Comment:
As you can see above sling used to read the entire response but we have a few response bodies measured in the GBs and really anything over a few dozen characters is too long for an error message. This is the improvement there.

Reply:
Consider using the io.LimitReader here